pkg/core/extension_store.go

```diff
@@ -292,7 +292,7 @@ func (store *ExtensionStore) getExtensionRecord(ctx context.Context, extensionID
 
 // GetAllExtensionss - retrieve all registered extensions, with all participating services, endpoints and credentials
 func (store *ExtensionStore) GetAllExtensions(ctx context.Context) (result []*domain.ExtensionRecord, err error) {
-	result = make([]*domain.ExtensionRecord, len(store.items))
+	result = make([]*domain.ExtensionRecord, 0)
```

Comment:
Why was this change needed?

Reply:
There was previously no method to allow listing all extensions in the registry/store. This is needed to implement GET /extensions

Reply:
Ah, this comment might be confusing. I was referring to the change between first and second commit:
```
-	result = make([]*domain.ExtensionRecord, len(store.items))
+	result = make([]*domain.ExtensionRecord, 0)
```

Reply:
Oh, I see. Well, you either initialize the slice to the intended size and use indexing (e.g. `result[i] = value`) or you initialize it to zero size and use `append`. In this case, I fixed a bug that gave me a bunch of `nil` values as the first entries.

Reply:
What about initializing it with 0 size but `len(store.items)` capacity? You could still use `append` there.

Reply:
(but that's not necessary to change)

Reply:
done